scripts/check-docs-quality.js

```diff
@@ -15,54 +14,76 @@
  * limitations under the License.
  */
 const { execSync, spawnSync } = require('child_process');
-const path = require('path');
-
-const listFilesTrackedByGit = 'git ls-files';
+// eslint-disable-next-line import/no-extraneous-dependencies
+const commandExists = require('command-exists');
 
 const inheritStdIo = {
   stdio: 'inherit',
 };
 
-const ERROR_MESSAGE =
-  'Please install vale linter(https://docs.errata.ai/vale/install). Ignore this message if already installed.\n';
+const LINT_SKIPPED_MESSAGE =
+  'Skipping documentation quality check (vale not found). Install vale linter (https://docs.errata.ai/vale/install) to enable.\n';
+const LINT_ERROR_MESSAGE = `Language linter (vale) generated errors. Please check the errors and review any markdown files that you changed.
+  Possibly update .github/styles/vocab.txt to add new valid words.\n`;
+const VALE_NOT_FOUND_MESSAGE = `Language linter (vale) was not found. Please install vale linter (https://docs.errata.ai/vale/install).\n`;
 
-// xargs is not supported by shx.
-if (process.platform === 'win32') {
-  const validMDFilesCommand = `${listFilesTrackedByGit} | .\\node_modules\\.bin\\shx grep ".md"`;
-  try {
-    // get list of all md files except in directories of gitignore.
-    let filesToLint = execSync(validMDFilesCommand, {
-      stdio: ['ignore', 'pipe', 'inherit'],
-    });
+// Note: Make sure the script is run as `node check-docs-quality.js [FILES]` instead of `./check-docs-quality.js [FILES]`
+// If the script receives arguments (file paths), the script is run exclusively on them. (e.g. when run via pre-commit hook)
+const getFilesToLint = () => {
+  // Files have been provided as arguments
+  if (process.argv.length > 2) {
+    return process.argv.slice(2);
+  }
 
-    // set all file(s) path as absolute path
-    filesToLint = filesToLint
-      .toString()
-      .split('\n')
-      .map(filepath => (filepath ? path.join(process.cwd(), filepath) : null))
-      .filter(Boolean);
+  let command = `git ls-files | ./node_modules/.bin/shx grep ".md"`;
+  if (process.platform === 'win32') {
+    command = `git ls-files | .\\node_modules\\.bin\\shx grep ".md"`;
+  }
 
-    const output = spawnSync('vale', filesToLint, inheritStdIo);
+  return execSync(command, {
+    stdio: ['ignore', 'pipe', 'inherit'],
+  })
+    .toString()
+    .split('\n');
```

Comment:
Are these relative paths working in windows? Just wants to confirm.

Reply:
I am not sure to be honest. If you can test this branch on Windows, it will be amazing!

Although, our contributors (on any OS) probably would not encounter this line since `lint-staged` will provide the absolute path to those files. 💯